### PR TITLE
LPS-187024 Adds AI Creator toolbar to the CKEditor when is enabled

### PR DESCRIPTION
**Pull request by lfbesada**
https://issues.liferay.com/browse/LPS-187024

Hi Team ✋ 

From the Echo Team, we are working on a story to [Provide a button to trigger AI in the web content editor.](https://issues.liferay.com/browse/LPS-181285) As part of that story, we need to add an attribute that indicates if the toolbar should be added or not. We are passing it to the editor configuration in a pull request that is reviewing by the Forms team, please see: https://github.com/liferay-forms/liferay-portal/pull/2238

In this pull request, I am adding the AI-Creator toolbar when this setting is true.

Could you please review it?

Thanks in advance! ❤️

### CODE REVIEW

File: modules/apps/frontend-editor/frontend-editor-ckeditor-web/src/main/java/com/liferay/frontend/editor/ckeditor/web/internal/editor/configuration/CKEditorConfigContributor.java
Comment (dsanz) [inline modules/apps/frontend-editor/frontend-editor-ckeditor-web/src/main/java/com/liferay/frontend/editor/ckeditor/web/internal/editor/configuration/CKEditorConfigContributor.java:177]: We can avoid passing custom flags if we apply the same pattern as `isShowSource()` method, that is, encapsulating the logic into a method:

```
boolean showAICreator(Map<String, Object> inputEditorTaglibAttributes) {
return GetterUtil.getBoolean(
			inputEditorTaglibAttributes.get(
				CKEditorConstants.ATTRIBUTE_NAMESPACE + ":showAICreator"));
}
```
Then using it from the places where we calculate the toolbars, as info is already available in `inputEditorTaglibAttributes` 

In addition, if  `isShowAICreator()` is moved to the the parent class, we'll give a chance to other editor config contributors inheriting from `BaseCKEditorConfigContributor` (not just the CKEditor one) to check and add that button to their respective toolbars. Not sure if you need this now, though.

**Review by dsanz — CHANGES_REQUESTED** [review]
Thanks @lfbesada :) 
Overall LGTM, just added a minor suggestion to apply an existing pattern and (optiionally) make that button available to other editors

**Comment by lfbesada** [comment]
ci:test:sf

**Comment by lfbesada** [comment]
ci:test:relevant

**Review by dsanz — APPROVED** [review]
LGTM, ready to fw once tests are green

@mateusralv pls let us know if failing tests could be related to this.

**Comment by mateusralv** [comment]
Hi @lfbesada and @dsanz the failures are not related. 

[Testray Comparison](https://testray.liferay.com/home/-/testray/runs/compare?formDate=1686572376994&testrayRunIdA=2345761880&testrayRunIdB=2346780561)

**Comment by lfbesada** [comment]
Hi @dsanz @mateusralv 
Thank you very much for the review and the analysis of the test failures ❤️ 
Can you please guide me on the next steps?
Should I launch again the relevant tests or should I do the manual forward to Brian or does your team handle it?
Thank you very much!
Regards

**Comment by bryceosterhaus** [comment]
ci:forward:force

**Comment by bryceosterhaus** [comment]
@lfbesada I went ahead and forwarded it. For the future, we just require a review approval(Dani gave that) and then CI passing or QA sign off(@mateusralv did that [here](https://github.com/liferay-frontend/liferay-portal/pull/3424#issuecomment-1587297331))

**Comment by lfbesada** [comment]
Hi @bryceosterhaus 
Thanks a lot! ❤️ 
To take this into account in future PRs, because each repo has its particularities 😅, once the review and CI approval are done, should the author manually PR Brian if the forward hasn't worked or the Team handles it?
Regards

**Comment by bryceosterhaus** [comment]
@lfbesada if the forward doesn't work, you can manually forward to Brian and just link to where QA gave approval.

So for this one, forward seems to not be working, feel free to manually send

**Pull request closed without merging**
